src/dtclient/kubernetes_settings.go

```diff
@@ -0,0 +1,251 @@
+package dtclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type postKubernetesSettings struct {
+	Label                           string `json:"label"`
+	ClusterIdEnabled                bool   `json:"clusterIdEnabled"`
+	ClusterId                       string `json:"clusterId"`
+	CloudApplicationPipelineEnabled bool   `json:"cloudApplicationPipelineEnabled"`
+	OpenMetricsPipelineEnabled      bool   `json:"openMetricsPipelineEnabled"`
+	Enabled                         bool   `json:"enabled"`
+	EventProcessingActive           bool   `json:"eventProcessingActive"`
+	EventProcessingV2Active         bool   `json:"eventProcessingV2Active"`
+	FilterEvents                    bool   `json:"filterEvents"`
+}
+
+type postKubernetesSettingsBody struct {
+	SchemaId      string                 `json:"schemaId"`
+	SchemaVersion string                 `json:"schemaVersion"`
+	Scope         string                 `json:"scope,omitempty"`
+	Value         postKubernetesSettings `json:"value"`
+}
+
+type monitoredEntitiesResponse struct {
+	TotalCount int               `json:"totalCount"`
+	PageSize   int               `json:"pageSize"`
+	Entities   []MonitoredEntity `json:"entities"`
+}
+
+type MonitoredEntity struct {
+	EntityId    string `json:"entityId"`
+	DisplayName string `json:"displayName"`
+	LastSeenTms int64  `json:"lastSeenTms"`
+}
+
+type GetSettingsResponse struct {
+	TotalCount int `json:"totalCount"`
+}
+
+type postSettingsResponse struct {
+	ObjectId string `json:"objectId"`
+}
+
+type getSettingsErrorResponse struct {
+	ErrorMessage getSettingsError `json:"error"`
+}
+
+type getSettingsError struct {
+	Code                 int
+	Message              string
+	ConstraintViolations constraintViolations
+}
+
+type constraintViolations []struct {
+	ParameterLocation string
+	Location          string
+	Message           string
+	Path              string
+}
+
+func (dtc *dynatraceClient) CreateKubernetesSetting(name, kubeSystemUUID, scope string) (string, error) {
+	if name == "" {
+		return "", errors.New("no name given")
+	}
+	if kubeSystemUUID == "" {
+		return "", errors.New("no kube-system namespace UUID given")
+	}
+
+	body := []postKubernetesSettingsBody{
+		{
+			SchemaId:      "builtin:cloud.kubernetes",
+			SchemaVersion: "1.0.27",
```

Comment:
does this schemaversion ever change?

Reply:
Should not change often, but the version should be stated, so that the API can try to migrate to the newest version.
Otherwise the API assumes the newest version, which might be incompatible with the given schema.

This should still be updated accordingly when the Kubernetes settings schema version changes to mitigate the risk of incompatibilities.